File: 프로그래머스/Lv1/소수 만들기.py
# 내 풀이
def solution(nums):
    answer = 0

    for i in range(len(nums)-2):
        for j in range(i+1, len(nums)-1):
            for k in range(j+1, len(nums)):
                if(decimal(nums[i]+nums[j]+nums[k]) == 0):
                    answer += 1

    return answer

def decimal(num):
    cnt = 0
    
    for i in range(2, num):
        if num % i == 0:
            cnt += 1
    
    return cnt

# 다른 사람 풀이
def solution2(nums):
    from itertools import combinations as cb 
    answer = 0
    for a in cb(nums, 3): # combinations(iterable, r) : iterable에서 원소 개수가 r개인 조합 뽑기
        cand = sum(a)
        for j in range(2, cand):
            if cand%j==0:
                break
        else:
            answer += 1
    return answer

print(solution([1,2,3,4]))
print(solution2([1,2,7,6,4]))

File: 프로그래머스/Lv1/소수 만들기.js
// 내 풀이
function solution(nums) {
    let answer = 0;
    
    for(let i=0; i<nums.length-2; i++){
        for(let j=i+1; j<nums.length-1; j++){
            for(let k=j+1; k<nums.length; k++){
                if(decimal(nums[i]+nums[j]+nums[k]) === 0){
                    answer++;
                }
            }
        }
    }
    
    return answer;
}

function decimal(num){
    let cnt = 0;
    
    for(let i=2; i<=num**0.5; i++){
        num%i === 0 ? cnt++ : 0
    }
    
    return cnt;
}

console.log(solution([1,2,3,4]));
console.log(solution([1,2,7,6,4]));